### PR TITLE
V0.1

### PR DESCRIPTION
**Pull request by krzysztofsmokowski**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class PublicNeeds(object):
+    def __init__(self):
+        pass
+
+    def requesting(self, page=1, limit=1000, last_page=4):
+        '''
+        Kwarg page will be used in for to iterate from this number to last_page
+        Limit is also used with default value of 100 as recommended in api docs.
+        For now i willl use self.requested_pages to append
```
Comment (Ogrodniczek) [inline app.py]: please use english in your comments. This seems to be unfinished docstring.

File: app.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        '''
+        requested_pages = []
+        for dictionary in range(page, last_page):
```
Comment (Ogrodniczek) [inline app.py]: dictionary is meaningless

File: app.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            req = requests.get('https://api-v3.mojepanstwo.pl/dane/zamowienia_publiczne.json?page={}&limit={}'.format(page, limit)).json()
```
Comment (Ogrodniczek) [inline app.py]: move it to separate method/function, extract url.

File: app.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import requests
+
+class PublicNeeds(object):
+    def __init__(self):
+        pass
+
+    def requesting(self, page=1, limit=1000, last_page=4):
```
Comment (Ogrodniczek) [inline app.py]: wrong function name

File: app.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        Kwarg page will be used in for to iterate from this number to last_page
+        Limit is also used with default value of 100 as recommended in api docs.
+        For now i willl use self.requested_pages to append
+        '''
+        requested_pages = []
+        for dictionary in range(page, last_page):
+            req = requests.get('https://api-v3.mojepanstwo.pl/dane/zamowienia_publiczne.json?page={}&limit={}'.format(page, limit)).json()
+            page+=1
+            requested_pages.append(req)
+            return(requested_pages)
```
Comment (Ogrodniczek) [inline app.py]: this function will end after one iteration, is it expected behavior?

File: app.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+    def service_information(self, region):
+        requested_jsons = self.requesting()
+        without_executor = []
+        for single_json in requested_jsons:
```
Comment (Ogrodniczek) [inline app.py]: meaningless names

File: app.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            for orders in single_json["Dataobject"]:
+                if orders["data"]["zamowienia_publiczne.zamawiajacy_wojewodztwo"] == region:
+                    print(orders["data"]["zamowienia_publiczne.nazwa"])
+                    print(orders["data"]["zamowienia_publiczne_rodzaje.nazwa"])
+                    if orders["data"]["zamowienia_publiczne.wykonawca_str"] == "":
+                        print("brak wykonawcy")
+                    else:
+                        print(orders["data"]["zamowienia_publiczne.wykonawca_str"])
+                    print("")
```
Comment (Ogrodniczek) [inline app.py]: seems to be not needed

File: app.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                    print(orders["data"]["zamowienia_publiczne.zamawiajacy_wojewodztwo"])
+
+def main():
+    print('done')
```
Comment (Ogrodniczek) [inline app.py]: debug print, not needed.

File: app.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    pubs = PublicNeeds()
+    #print(pubs.requesting())
```
Comment (Ogrodniczek) [inline app.py]: remove

File: app.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+                    if orders["data"]["zamowienia_publiczne.wykonawca_str"] == "":
+                        print("brak wykonawcy")
+                    else:
+                        print(orders["data"]["zamowienia_publiczne.wykonawca_str"])
+                    print("")
+                    print(orders["data"]["zamowienia_publiczne.zamawiajacy_wojewodztwo"])
+
+def main():
+    print('done')
+    pubs = PublicNeeds()
```
Comment (Ogrodniczek) [inline app.py]: pubs could be replaced with better name

File: flaskapp.py [hunk 11]
```diff
@@ -0,0 +1,17 @@
+from flask import Flask, json
+from flask_restful import reqparse, abort, Api, Resource
+from app import PublicNeeds
```
Comment (Ogrodniczek) [inline flaskapp.py:3]: app should be flask app, and the PublicNeeds should be included in module file called differently than app.py

File: flaskapp.py [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+class Region(Resource):
+    def get(self, region):
+        pubs = PublicNeeds()
+        data = pubs.service_information(region=region)
+        return data, 200
+
+
+api.add_resource(Region, '/region/<region>')
+if __name__ == '__main__':
+    app.run(debug=False, host='0.0.0.0', port=7600)
```
Comment (Ogrodniczek) [inline flaskapp.py:17]: maybe change default port to 80.

**Review by Ogrodniczek — CHANGES_REQUESTED** [review]
- Please add proper Dockerfile.
- Update requirements.txt
- change current app.py to some module name and flaskapp.py to app.py

File: app.py (hunk 4 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: app.py (hunk 1 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed and done

File: app.py (hunk 5 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: app.py (hunk 8 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: app.py (hunk 7 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: app.py (hunk 6 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: app.py (hunk 10 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: app.py (hunk 2 above)
Comment (krzysztofsmokowski) [reply app.py]: fixed

File: flaskapp.py (hunk 12 above)
Comment (krzysztofsmokowski) [reply flaskapp.py:17]: Another service is occupying this port

**Pull request merged**